specs/calculator.spec.js

```diff
@@ -0,0 +1,148 @@
+            queue.push(top, secondEl);
+            queue.pop();
+            expect(queue.top()).to.be.equal(secondEl);
+        });
+
+        it('should be add to queue after push if queue is empty', function() {
+            var size = 5;
+            var queue = new Queue(size);
+            var top = 1;
+            queue.push(top);
+            expect(queue.top()).to.be.equal(top);
+        });
+    });
+
+    describe('merge', function() {
+        it('should merge second queue at the end of first queue', function() {
+            var size = 5;
+            var firstArr = [1, 2, 3];
+            var secondArr = [4, 5, 6];
+            var firstQueue = new Queue(size);
+            var secondQueue = new Queue(size);
+            firstQueue.push(...firstArr);
+            secondArr.push(...secondArr);
+            var doubleQueue = firstQueue.merge(secondQueue, false);
+            expect(doubleQueue).to.be.equal([...firstArr, ...secondArr]);
+        });
+
+        it('should merge first queue at the end of second queue', function() {
+            var size = 5;
+            var firstArr = [1, 2, 3];
+            var secondArr = [4, 5, 6];
+            var firstQueue = new Queue(size);
+            var secondQueue = new Queue(size);
+            firstQueue.push(...firstArr);
+            secondArr.push(...secondArr);
+            var doubleQueue = firstQueue.merge(secondQueue, true);
+            expect(doubleQueue).to.be.equal([...secondArr, ...firstArr]);
+        });
+
+        it('should set top of first queue', function() {
+            var size = 5;
+            var firstArr = [1, 2, 3];
+            var secondArr = [4, 5, 6];
+            var firstQueue = new Queue(size);
+            var secondQueue = new Queue(size);
+            firstQueue.push(...firstArr);
+            secondArr.push(...secondArr);
+            var doubleQueue = firstQueue.merge(secondQueue, false);
+            expect(doubleQueue.top()).to.be.equal(1);
+        });
+
+        it('should set top of second queue', function() {
+            var size = 5;
+            var firstArr = [1, 2, 3];
+            var secondArr = [4, 5, 6];
+            var firstQueue = new Queue(size);
+            var secondQueue = new Queue(size);
+            firstQueue.push(...firstArr);
+            secondArr.push(...secondArr);
+            var doubleQueue = firstQueue.merge(secondQueue, true);
+            expect(doubleQueue.top()).to.be.equal(4);
+        });
+
+        it('new queue should be instance of Queue', function() {
+            var size = 5;
+            var firstArr = [1, 2, 3];
+            var secondArr = [4, 5, 6];
+            var firstQueue = new Queue(size);
+            var secondQueue = new Queue(size);
+            firstQueue.push(...firstArr);
+            secondArr.push(...secondArr);
+            var doubleQueue = firstQueue.merge(secondQueue, true);
+            expect(doubleQueue instanceof Queue).to.be.equal(true);
+        });
+    });
+});
```

Comment:
issue for all tests: use not calculated but hardcoded values as it's more readable.
not enough tests with negative scenarios.